Packs/CiscoEmailSecurity/Integrations/CiscoEmailSecurity/CiscoEmailSecurity.py

```diff
@@ -0,0 +1,639 @@
+            timeout=2000
+        )
+
+    def list_entries_delete(self, list_type, request_body):
+        return self._http_request(
+            method='DELETE',
+            url_suffix=f"/sma/api/v2.0/quarantine/{list_type}",
+            json_data=request_body,
+            timeout=2000
+        )
+
+
+def parse_dates_to_ces_format(date_str):
+    splitted_date = date_str.split('T')
+    date_day = splitted_date[0]
+    full_date_time = splitted_date[1]
+    date_time_without_ms = full_date_time.split('.')[0]
+    date_time_without_seconds = date_time_without_ms[:-3]
+    date_time_with_zero_seconds = f'{date_time_without_seconds}:00.000Z'
+
+    return f'{date_day}T{date_time_with_zero_seconds}'
+
+
+def get_dates_for_test_module():
+    now = datetime.now()
+    start = now - timedelta(days=2)
+    end = now - timedelta(days=1)
+
+    start_date = start.isoformat()
+    end_date = end.isoformat()
+
+    start_date = parse_dates_to_ces_format(start_date)
+    end_date = parse_dates_to_ces_format(end_date)
+    return start_date, end_date
+
+
+def test_module(client: Client) -> str:
+    start_date, end_date = get_dates_for_test_module()
+    suffix_url = f"?startDate={start_date}&" \
+                 f"endDate={end_date}&ciscoHost=All_Hosts&searchOption=messages&offset=0&limit=20"
+    try:
+        client.list_messages(suffix_url)
+    except DemistoException as e:
+        if 'Forbidden' in str(e):
+            return 'Authorization Error: make sure API Key is correctly set'
+        else:
+            raise e
+    return 'ok'
+
+
+def date_to_cisco_date(date):
+    return date.replace(' ', 'T') + '.000Z'
+
+
+def set_limit(limit):
+    return int(limit) if limit and int(limit) <= MAX_MESSAGES_TO_GET else MAX_MESSAGES_TO_GET
+
+
+def build_url_params_for_list_report(args, report_counter):
+    start_date = date_to_cisco_date(args.get('start_date'))
+    end_date = date_to_cisco_date(args.get('end_date'))
+    url_params = f'/{report_counter}?startDate={start_date}&endDate={end_date}&device_type=esa'
+    return url_params
+
+
+def set_var_to_output_prefix(report_counter):
+    list_counter_words = report_counter.split('_')
+    counter_words = ''
+    for word in list_counter_words:
+        counter_words += word + ' '
+    counter_words_capital_letter = counter_words.title()
+    counter_output_prefix = counter_words_capital_letter.replace(' ', '')
+    return counter_output_prefix
+
+
+def list_report_command(client: Client, args: Dict[str, Any]):
+    counter = args.get('report_counter')
+    url_params = build_url_params_for_list_report(args, counter)
+    report_response_data = client.list_report(url_params)
+    report_data = report_response_data.get('data', {}).get('resultSet', [None])[0]
+    counter_output_prefix = set_var_to_output_prefix(counter)
+    return CommandResults(
+        readable_output=f'{report_response_data}',
+        outputs_prefix=f'CiscoEmailSecurity.Report.{counter_output_prefix}',
```

Comment:
why is the prefix being randomized?

Reply:
In report command, there are five things to report and I thought everyone needed a different path in context, I'm not right?
Should everyone be under report?